aioshelly/ble/const.py

```diff
@@ -10,31 +10,63 @@
 VAR_EVENT_TYPE = "%event_type%"
 VAR_ACTIVE = "%active%"
 VAR_VERSION = "%version%"
-VAR_INTERVAL_MS = "%interval_ms%"
-VAR_WINDOW_MS = "%window_ms%"
-VAR_DURATION_MS = "%duration_ms%"
-
-DEFAULT_INTERVAL_MS = 320
-DEFAULT_WINDOW_MS = 30
-DEFAULT_DURATION_MS = -1
 
 BLE_CODE = """
-// aioshelly BLE script 1.0
-BLE.Scanner.Subscribe(function (ev, res) {
-    if (ev === BLE.Scanner.SCAN_RESULT) {
-        Shelly.emitEvent("%event_type%", [
-            %version%,
-            res.addr,
-            res.rssi,
-            btoa(res.advData),
-            btoa(res.scanRsp)
-        ]);
+// aioshelly BLE script 2.0
+const queueServeTimer = 450; // in ms, timer for events emitting
+const burstSendCount =  5; // number if events, emitted on timer event
+const maxQueue =  80; // if the queue exceeds the limit, all new events are ignored until it empties
+
+let queue = [];
+let timerHandler = null;
+
+function timerCallback() {  
+  Timer.clear(timerHandler);
+  
+  for(let i = 0; i < burstSendCount; i++) {
+    if (queue.length <= 0) {
+      break;
     }
-});
-BLE.Scanner.Start({
-    duration_ms: %duration_ms%,
+
+    Shelly.emitEvent("%event_type%", queue[0]);
```

Comment:
Can we change the event to include multiple payloads at once?  That should reduce the amount of small packets the device has to send

Reply:
In ESPHome we do `Bluetooth advertisement bundling` which means we try to send 16 (32 if the device has SPIRAM at a time to minimize the time the wifi radio is on.

If you can do something similar here, I can take care of refactoring the integration code to do something similar to https://github.com/Bluetooth-Devices/bleak-esphome/blob/b93b9eec29b1183e72ffb5a0510891b328076743/src/bleak_esphome/backend/scanner.py#L40

Reply:
That also means we could update https://www.home-assistant.io/integrations/bluetooth/#remote-adapters-bluetooth-proxies to list Bluetooth advertisement bundling support for shelly

Reply:
With https://github.com/home-assistant-libs/aioshelly/pull/478/commits/619ce095c29ebf8c94372267f22e96fda5830699 the script will send data as an object now with 2 props, first one is "ver" - response version and the second one "advs" which is a list of arrays with the same data as before. 

The parser method `parse_ble_scan_result_event` now returns list with objects, and each object contains "address", "rssi" and "adv" keys.

Reply:
@bdraco Should I open a MR in the core repo to update the handler?

Reply:
I took care of it in https://github.com/home-assistant/core/pull/107917